### PR TITLE
Document oidc-agent profile reload (SOFTWARE-4866)

### PR DESCRIPTION
**Pull request by mtru32**

### CODE REVIEW

File: docs/software/requesting-tokens.md [hunk 1]
```diff
@@ -26,104 +26,139 @@ and refresh tokens from OpenID Connect token providers.
 
 ### Using a container
 
-1. Start an agent container in the background and name it `my-agent` to easily run subsequent commands against it:
+1. Registering a new profile:
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:29]:
```suggestion
### Registering an OIDC profile
```

File: docs/software/requesting-tokens.md [hunk 2]
```diff
@@ -26,104 +26,139 @@ and refresh tokens from OpenID Connect token providers.
...
 
-    1. On the next page, verify the scopes and client profile name, and click "Authorize".
+        1. On the next page, verify the scopes and client profile name, and click "Authorize".
 
-    1. Enter a password to encrypt your local client profile.
+        1. Enter a password to encrypt your local client profile.
        You'll need to remember this if you want to re-use this profile in subsequent sessions.
 
-1. Request a token using the client name that you used above with `oidc-gen`:
+1. Request a token using the client profile that you used above with `oidc-gen`:
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:69]:
```suggestion
### Requesting access tokens
```

File: docs/software/requesting-tokens.md [hunk 3]
```diff
@@ -26,104 +26,139 @@ and refresh tokens from OpenID Connect token providers.
...
+
+    !!! note
+        You must first [register a new profile](#using-a-container)
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:72]: When reorganizing, make sure to point to the corresponding profile registration section

File: docs/software/requesting-tokens.md [hunk 4]
```diff
@@ -26,104 +26,139 @@ and refresh tokens from OpenID Connect token providers.
...
-   For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to `<CE FQDN>:<CE PORT>`.
 
-1. Copy the output of `oidc-token` into a file on the host where you need SciToken authentication, e.g. an HTCondor or
+    1. For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to  
+    `<CE FQDN>:<CE PORT>`.
+
+        1. Copy the output of `oidc-token` into a file on the host where you need SciToken authentication, e.g. an HTCondor or
    XRootD client.
 
+1. Reload a profile:
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:86]:
```suggestion
### Reloading an OIDC profile
```

**Review by brianhlin — CHANGES_REQUESTED** [review]
For easier navigation, I'd like to see each of registration, requesting access tokens, and reloading a profile as separate sub-sections of each installation method (container vs RPM). So something like this:

```
## Before Starting
## Requesting Tokens Using a Container
### Registering an OIDC profile
### Requesting access tokens
### Reloading an OIDC profile
## Requesting Tokens with an RPM installation
### Registering an OIDC profile
### Requesting access tokens
### Reloading an OIDC profile
## Troubleshooting Tokens
```

File: docs/software/requesting-tokens.md [hunk 5]
```diff
@@ -18,149 +17,150 @@ Before you can request the appropriate tokens, you must have the following:
...
-            :::console
-            docker exec -it my-agent oidc-token --aud="<SERVER AUDIENCE>" <CLIENT PROFILE>
+1. Request a token using the client profile that you used with `oidc-gen`:
+	
+        :::console
+        docker exec -it my-agent oidc-token --aud="<SERVER AUDIENCE>" <CLIENT PROFILE>
 
 
-    1. For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to  
+For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:76]:
```suggestion
    For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to  
```

This text needs to be indented by 4 spaces to be associated with the above step

File: docs/software/requesting-tokens.md [hunk 6]
```diff
@@ -18,149 +17,150 @@ Before you can request the appropriate tokens, you must have the following:
...
-    1. Request token:
+1. Request a token using the client profile that you used with `oidc-gen`:
 
-            :::console
-            oidc-token --aud="<SERVER AUDIENCE>" <CLIENT PROFILE>
+        :::console
+        oidc-token --aud="<SERVER AUDIENCE>" <CLIENT PROFILE>
 
-    1. For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to  
+For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:145]:
```suggestion
    For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to  
```

File: docs/software/requesting-tokens.md [hunk 7]
```diff
@@ -18,149 +17,150 @@ Before you can request the appropriate tokens, you must have the following:
...
-        1. Copy the output of `oidc-token` into a file on the host where you need SciToken authentication, e.g. an HTCondor or
+  1. Copy the output of `oidc-token` into a file on the host where you need SciToken authentication, e.g. an HTCondor or
    XRootD client.
 
-1. Reload a profile:
+### Reloading an OIDC profile
 
-    1. Required if you restart the running container:
+!!! note
+    Required after restarting the running container.
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:85]: This also requires an existing profile so we should link to the relevant profile registration like the "Requesting access tokens" note

File: docs/software/requesting-tokens.md [hunk 8]
```diff
@@ -18,149 +17,150 @@ Before you can request the appropriate tokens, you must have the following:
...
 
-        1. Copy the output of `oidc-token` into a file on the host where you need SciToken authentication, e.g. an HTCondor or
+  1. Copy the output of `oidc-token` into a file on the host where you need SciToken authentication, e.g. an HTCondor or
    XRootD client.
 
-1. Reload a profile:
+### Reloading an OIDC profile
+
+!!! note
+    Required if you log out of the running machine
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:154]: This also requires an existing profile so we should link to the relevant profile registration like the "Requesting access tokens" note

File: docs/software/requesting-tokens.md [hunk 9]
```diff
@@ -18,149 +17,150 @@ Before you can request the appropriate tokens, you must have the following:
...
 
-1. Reload a profile:
+### Reloading an OIDC profile
 
-    1. Required if you restart the running container:
+!!! note
+    Required after restarting the running container.
 
-            docker exec -it my-agent oidc-add <CLIENT PROFILE>
+1. Reload profile:
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:87]: We should add an explicit step here that says something like

```
1.  If your existing container is not already running, start it:

        :::console
        docker start my-agent
```

File: docs/software/requesting-tokens.md [hunk 10]
```diff
@@ -18,149 +17,150 @@ Before you can request the appropriate tokens, you must have the following:
...
    XRootD client.
 
-1. Reload a profile:
+### Reloading an OIDC profile
+
+!!! note
+    Required if you log out of the running machine
 
-    1. Required if you log out of the running machine:
+1. Reload profile:
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:156]: We should add an explicit step here that says something like

```
1.  If you do not already have a running `oidc-agent`, start one:

        :::console
        eval `oidc-agent`
```

**Review by brianhlin — CHANGES_REQUESTED** [review]
Looking better but there are still some minor issues

**Comment by matyasselmeci** [comment]
Can you drop a link in https://opensciencegrid.atlassian.net/browse/SOFTWARE-4885 when this is done?

File: docs/software/requesting-tokens.md [hunk 11]
```diff
@@ -64,17 +62,44 @@ and refresh tokens from OpenID Connect token providers.
...
+
+1. Reload profile:
+
+            :::console
+            docker exec -it my-agent oidc-add <CLIENT PROFILE>
+
+    1. If your existing container is not already running, start it:
+
+                :::console
+                docker start my-agent
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:95]: The ordering of the steps is broken here. The first command will fail if your container isn't running!

Also, the indentation will cause the `docker start my-agent` step to be a sub-bullet of the first and we don't want that

File: docs/software/requesting-tokens.md [hunk 12]
```diff
@@ -114,18 +138,40 @@ and refresh tokens from OpenID Connect token providers.
...
+
+1. Reload profile:
+
+        :::console
+        oidc-add <CLIENT PROFILE>
+
+    1. If you do not already have a running 'oidc-agent', start one:
+
+            :::console
+            eval 'oidc-agent'
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:170]: Same problem here with the ordering and indentation

File: docs/software/requesting-tokens.md [hunk 13]
```diff
@@ -114,18 +138,40 @@ and refresh tokens from OpenID Connect token providers.
...
         :::console
-        oidc-token --aud="<SERVER AUDIENCE>" <CLIENT NAME>
+        oidc-token --aud="<SERVER AUDIENCE>" <CLIENT PROFILE>
 
-   For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to `<CE FQDN>:<CE PORT>`.
+    For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to  
+    `<CE FQDN>:<CE PORT>`.
 
-1. Copy the output of `oidc-token` into a file on the host where you need SciToken authentication, e.g. an HTCondor or
+  1. Copy the output of `oidc-token` into a file on the host where you need SciToken authentication, e.g. an HTCondor or
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:154]: Extra space seems to have snuck in

File: docs/software/requesting-tokens.md [hunk 14]
```diff
@@ -64,17 +62,44 @@ and refresh tokens from OpenID Connect token providers.
...
-        docker exec -it my-agent oidc-token --aud="<SERVER AUDIENCE>" <CLIENT NAME>
+        docker exec -it my-agent oidc-token --aud="<SERVER AUDIENCE>" <CLIENT PROFILE>
+
 
-   For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to `<CE FQDN>:<CE PORT>`.
+    For tokens used against an HTCondor-CE, set `<SERVER AUDIENCE>` to  
+    `<CE FQDN>:<CE PORT>`.
 
-1. Copy the output of `oidc-token` into a file on the host where you need SciToken authentication, e.g. an HTCondor or
+  1. Copy the output of `oidc-token` into a file on the host where you need SciToken authentication, e.g. an HTCondor or
```
Comment (brianhlin) [inline docs/software/requesting-tokens.md:79]: An extra space snuck in

**Review by brianhlin — CHANGES_REQUESTED** [review]
Almost there but there are still a few changes that are needed

**Review by brianhlin — APPROVED** [review]
LGTM

**Comment by brianhlin** [comment]
CI failure unrelated, merging

**Pull request merged**